modules/biobank/sql/biobankSchema.sql

```diff
@@ -0,0 +1,450 @@
+  CONSTRAINT `FK_biobank_container_ContainerTypeID`
+    FOREIGN KEY (`ContainerTypeID`) REFERENCES `biobank_container_type`(`ContainerTypeID`)  
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `FK_biobank_container_ContainerStatusID`
+    FOREIGN KEY (`ContainerStatusID`) REFERENCES `biobank_container_status`(`ContainerStatusID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `FK_biobank_container_OriginCenterID`
+    FOREIGN KEY (`OriginCenterID`) REFERENCES `psc`(`CenterID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `FK_biobank_container_CurrentCenterID`
+    FOREIGN KEY (`CurrentCenterID`) REFERENCES `psc`(`CenterID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `UK_biobank_container_Barcode` UNIQUE (`Barcode`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+/*Specimen*/
+CREATE TABLE `biobank_specimen_type` (
+  `SpecimenTypeID` integer unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(50) NOT NULL,
+  `FreezeThaw` BIT(1) NOT NULL,
+  `ParentSpecimenTypeID` integer unsigned,
+  `Regex` varchar(100),
+  CONSTRAINT `PK_biobank_specimen_type` PRIMARY KEY (`SpecimenTypeID`),
+  CONSTRAINT `FK_biobank_specimen_type_ParentSpecimenTypeID`
+    FOREIGN KEY (`ParentSpecimenTypeID`) REFERENCES `biobank_specimen_type`(`SpecimenTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `UK_biobank_specimen_type_Label` UNIQUE (`Label`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `biobank_specimen_protocol` (
+  `SpecimenProtocolID` integer unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(50) NOT NULL,
+  `SpecimenTypeID` integer unsigned NOT NULL,
+  CONSTRAINT `PK_biobank_specimen_protocol` PRIMARY KEY (`SpecimenProtocolID`),
+  CONSTRAINT `FK_biobank_specimen_protocol_SpecimenTypeID`
+    FOREIGN KEY (`SpecimenTypeID`) REFERENCES `biobank_specimen_type`(`SpecimenTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `UK_biobank_specimen_protocol_Label` UNIQUE (`Label`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8; 
+
+CREATE TABLE `biobank_specimen_method` (
+  `SpecimenMethodID` integer unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(50) NOT NULL,
+  `SpecimenTypeID` integer unsigned NOT NULL,
+  CONSTRAINT `PK_biobank_specimen_method` PRIMARY KEY (`SpecimenMethodID`),
+  CONSTRAINT `FK_biobank_specimen_method_SpecimenTypeID`
+    FOREIGN KEY (`SpecimenTypeID`) REFERENCES `biobank_specimen_type`(`SpecimenTypeID`)
+    ON UPDATE RESTRICT ON DELETE RESTRICT,
+  CONSTRAINT `UK_biobank_specimen_method_Label` UNIQUE (`Label`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8; 
+
+CREATE TABLE `biobank_specimen` (
+  `SpecimenID` integer unsigned NOT NULL AUTO_INCREMENT,
+  `ContainerID` integer unsigned NOT NULL, /*Index by ContainerID*/
+  `SpecimenTypeID` integer unsigned NOT NULL,
+  `Quantity` DECIMAL(10, 3) NOT NULL,
+  `UnitID` integer unsigned NOT NULL,
+  `CandidateID` int(6) NOT NULL,
+  `SessionID` integer unsigned UNSIGNED NOT NULL,
```

Comment:
Interesting. I don't see why not since the `candidate` table can simply be joined on for the query. It will take a bit of re-working of dataflow, but should be feasible. 

@ridz1208 do you have any thoughts or apprehensions about this?

Reply:
done!